### DISCUSSION TITLE
multipage routing

### DISCUSSION THREAD

**Discussion by zillionare** [h2oai/wave#1382] [Q&A] [answered]
is this possible to write application with multiple pages/routes, and respond with other page's url?

from examples, 
```
from h2o_wave import Q, main, app, ui

@app('/toss')
async def serve(q: Q):
    q.page['sides'] = ui.form_card(
        box='1 1 4 4',
        items=[
            ui.button(name='#heads', label='Heads'),
            ui.button(name='#tails', label='Tails'),
        ],
    )
    await q.page.save()
```
sounds every handler just save page, and doesn't respond anything.

**Comment by lo5** [comment, marked as answer]
There's no support for registering multiple routes inside one app. A Wave server typically hosts multiple apps, and each app "listens" to one route.

For adding multiple "pages"/UIs to your app, add/remove cards to/from the page (I explained this in a reply to your other question https://github.com/h2oai/wave/discussions/1383#discussioncomment-2640935). 

If you want bookmark-able routes, use [hash routing](https://wave.h2o.ai/docs/routing#hash-routing).

**Comment by azim-b** [comment]
As a workaround, can you use the hash routing with paths to make a "multi-page" app?

Please see an example here and let me know if it helps:
https://github.com/h2oai/wave/discussions/1407#discussioncomment-2702281
